### PR TITLE
API provides registration possibility

### PR DESCRIPTION
**Pull request by clalil**
# API provides registration possibility
[PT-link](https://www.pivotaltracker.com/story/show/169253798)

## This PR contains
- Configuration and installation of devise token auth
- Request specs

## What have we learned
- Further experience of implementing devise token auth configuration and write specs

### CODE REVIEW

**Review by miyeaier — APPROVED** [review]

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# app/models/user.rb
+class User < ActiveRecord::Base
+  extend Devise::Models
+    # Include default devise modules.
```
Comment (piavW) [inline app/models/user.rb]: Remove comment :)

**Review by jonas-masharqa — APPROVED** [review]

**Review by piavW — APPROVED** [review]

**Review by tochman — CHANGES_REQUESTED** [review]
Even if we initially accepted multiple expectations in a test, we’ve also talked about avoiding using that in on several occasions.

File: app/models/user.rb (hunk 1 above)
Comment (clalil) [reply app/models/user.rb]: Of course! :)

**Review by AlexSaxena — APPROVED** [review]

File: spec/requests/v1/api_provides_registration_possibility.rb [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        expect(response_json['errors']['email']).to eq ['is not an email']
+      end
+
+      it 'returns error status' do
+        expect(response.status).to eq 422
+      end
+  end
+
+  describe 'an already registered email' do
+    FactoryBot.create(:user,
```
Comment (tochman) [inline spec/requests/v1/api_provides_registration_possibility.rb]: Please leave out the module name. You are including the FactoryBot methods in your rails_helper.

**Review by tochman — CHANGES_REQUESTED** [review]

**Review by AlexSaxena — APPROVED** [review]

File: spec/requests/v1/api_provides_registration_possibility.rb (hunk 2 above)
Comment (clalil) [reply spec/requests/v1/api_provides_registration_possibility.rb]: Of course, we have just removed it.

File: app/models/user.rb [hunk 3]
```diff
@@ -0,0 +1,6 @@
+class User < ActiveRecord::Base
+  extend Devise::Models
+    devise :database_authenticatable, :registerable,
+            :recoverable, :rememberable, :trackable, :validatable
+    include DeviseTokenAuth::Concerns::User
+  end
```
Comment (tochman) [inline app/models/user.rb]: Please fix formatting in this file

File: spec/requests/v1/api_provides_registration_possibility.rb [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+      it 'returns error status' do
+        expect(response.status).to eq 422
+      end
+  end
+
+  describe 'submits an already registered email' do
+    create(:user)
+    before do
+      post '/v1/auth',
+        params: { email: 'user@random.com',
```
Comment (tochman) [inline spec/requests/v1/api_provides_registration_possibility.rb]: If you use a specific value in your tests, you should set it when creating your object using a factory. I had to check the factory settings to understand where that specific email came from. It’s a small but IMO important detail.

**Review by tochman — APPROVED** [review]

File: app/models/user.rb (hunk 3 above)
Comment (clalil) [reply app/models/user.rb]: Thank you, we have adjusted.

File: spec/requests/v1/api_provides_registration_possibility.rb (hunk 4 above)
Comment (clalil) [reply spec/requests/v1/api_provides_registration_possibility.rb]: Agreed. Have changed this particular describe block, as well as fixed a bug.

File: app/models/user.rb (hunk 3 above)
Comment (tochman) [reply app/models/user.rb]: Not that I can see

**Pull request merged**